packages/grid/src/vaadin-grid-sort-column.d.ts

```diff
@@ -30,22 +21,13 @@ export interface GridSortColumnEventMap extends HTMLElementEventMap, GridSortCol
  *  <vaadin-grid-column>
  *    ...
  * ```
- *
- * @fires {CustomEvent} direction-changed - Fired when the `direction` property changes.
  */
-declare class GridSortColumn<TItem = GridDefaultItem> extends GridColumn<TItem> {
-  /**
-   * JS Path of the property in the item used for sorting the data.
-   */
-  path: string | null | undefined;
-
-  /**
-   * How to sort the data.
-   * Possible values are `asc` to use an ascending algorithm, `desc` to sort the data in
-   * descending direction, or `null` for not sorting the data.
-   */
-  direction: GridSorterDirection | null | undefined;
+declare class GridSortColumn<TItem = GridDefaultItem> extends HTMLElement {}
 
+interface GridSortColumn<TItem = GridDefaultItem>
+  extends GridSortColumnMixinClass,
+    GridColumnMixin<TItem, GridColumn<TItem>>,
```

Comment:
I was originally planning to use `GridSortColumn<TItem>` for the generic type here but unfortunately, that leads to TS conflict with the `renderer` params -> `column` types.

Now it basically works the same as on `main`: the column type passed from a renderer is just `GridColumn` (instead of `GridSortColumn`):

<img width="648" alt="Screenshot 2023-10-06 at 15 31 06" src="https://github.com/vaadin/web-components/assets/1222264/57685b1d-206d-41e1-bbfa-355ea7e3a2f4">